tools/trace_analyzer_tool.cc

```diff
@@ -1504,322 +1614,187 @@ Status TraceAnalyzer::Handle(const WriteQueryTraceRecord& record,
-    ta_[TraceOperationType::kSingleDelete].sample_count = 0;
-  }
-  if (ta_[TraceOperationType::kSingleDelete].sample_count > 0) {
-    ta_[TraceOperationType::kSingleDelete].sample_count++;
-    return Status::OK();
-  }
-  ta_[TraceOperationType::kSingleDelete].sample_count++;
-
-  if (!ta_[TraceOperationType::kSingleDelete].enabled) {
-    return Status::OK();
-  }
-  s = KeyStatsInsertion(TraceOperationType::kSingleDelete, column_family_id,
-                        key.ToString(), value_size, c_time_);
-  if (!s.ok()) {
-    return Status::Corruption("Failed to insert key statistics");
-  }
-  return s;
+  assert(write_result_ != nullptr);
+  write_result_->Reset(TraceOperationType::kSingleDelete, column_family_id, key,
+                       0);
+  return write_result_->Accept(this);
 }
 
 // Handle the DeleteRange request in the write batch of the trace
 Status TraceAnalyzer::DeleteRangeCF(uint32_t column_family_id,
                                     const Slice& begin_key,
                                     const Slice& end_key) {
-  Status s;
-  size_t value_size = 0;
-  if (FLAGS_convert_to_human_readable_trace && trace_sequence_f_) {
-    s = WriteTraceSequence(TraceOperationType::kRangeDelete, column_family_id,
-                           begin_key.ToString(), value_size, c_time_);
-    if (!s.ok()) {
-      return Status::Corruption("Failed to write the trace sequence to file");
-    }
-  }
-
-  if (ta_[TraceOperationType::kRangeDelete].sample_count >= sample_max_) {
-    ta_[TraceOperationType::kRangeDelete].sample_count = 0;
-  }
-  if (ta_[TraceOperationType::kRangeDelete].sample_count > 0) {
-    ta_[TraceOperationType::kRangeDelete].sample_count++;
-    return Status::OK();
-  }
-  ta_[TraceOperationType::kRangeDelete].sample_count++;
-
-  if (!ta_[TraceOperationType::kRangeDelete].enabled) {
-    return Status::OK();
-  }
-  s = KeyStatsInsertion(TraceOperationType::kRangeDelete, column_family_id,
-                        begin_key.ToString(), value_size, c_time_);
-  s = KeyStatsInsertion(TraceOperationType::kRangeDelete, column_family_id,
-                        end_key.ToString(), value_size, c_time_);
-  if (!s.ok()) {
-    return Status::Corruption("Failed to insert key statistics");
-  }
-  return s;
+  assert(write_result_ != nullptr);
+  write_result_->Reset(TraceOperationType::kRangeDelete,
+                       {column_family_id, column_family_id},
+                       {begin_key, end_key}, {0, 0}, {true, false});
+  return write_result_->Accept(this);
 }
 
 // Handle the Merge request in the write batch of the trace
 Status TraceAnalyzer::MergeCF(uint32_t column_family_id, const Slice& key,
                               const Slice& value) {
+  assert(write_result_ != nullptr);
+  write_result_->Reset(TraceOperationType::kMerge, column_family_id, key,
+                       value.size());
+  return write_result_->Accept(this);
+}
+
+Status TraceAnalyzer::Handle(const TraceAnalysisResult& result) {
```

Comment:
Need some comments here to explain what is the job of this "Handle".

Reply:
ACK.

It is used to handle the generic analysis result for outputting purpose. If we don't use `TraceAnalysisResult`, we can probably replace it with a single function that does the same job.